ql/src/test/results/clientpositive/llap/explainuser_4.q.out

```diff
@@ -119,24 +119,24 @@ Stage-0
           PARTITION_ONLY_SHUFFLE [RS_11]
             Group By Operator [GBY_10] (rows=1 width=8)
               Output:["_col0"],aggregations:["count()"]
-              Merge Join Operator [MERGEJOIN_29] (rows=4626 width=8)
+              Merge Join Operator [MERGEJOIN_29] (rows=3449 width=8)
                 Conds:RS_32._col0=RS_35._col0(Inner)
               <-Map 1 [SIMPLE_EDGE] vectorized, llap
                 SHUFFLE [RS_32]
                   PartitionCols:_col0
-                  Select Operator [SEL_31] (rows=3078 width=2)
+                  Select Operator [SEL_31] (rows=2297 width=2)
                     Output:["_col0"]
-                    Filter Operator [FIL_30] (rows=3078 width=2)
-                      predicate:cint BETWEEN 1000000 AND 3000000
+                    Filter Operator [FIL_30] (rows=2297 width=2)
+                      predicate:(cint is not null and cint BETWEEN 1000000 AND 3000000)
                       TableScan [TS_0] (rows=12288 width=2)
                         default@alltypesorc,a,Tbl:COMPLETE,Col:COMPLETE,Output:["cint"]
               <-Map 4 [SIMPLE_EDGE] vectorized, llap
                 SHUFFLE [RS_35]
                   PartitionCols:_col0
-                  Select Operator [SEL_34] (rows=2298 width=2)
+                  Select Operator [SEL_34] (rows=1715 width=2)
                     Output:["_col0"]
-                    Filter Operator [FIL_33] (rows=2298 width=8)
-                      predicate:(cint BETWEEN 1000000 AND 3000000 and cbigint is not null)
+                    Filter Operator [FIL_33] (rows=1715 width=8)
+                      predicate:(cint is not null and cint BETWEEN 1000000 AND 3000000 and cbigint is not null)
```

Comment:
It is interesting that the seemingly redundant `IS NOT NULL` predicate appears alongside the `BETWEEN` clause. It’s probably not gonna harm the performance much but it would be nice to see if it is something caused by Hive changes or due to Calcite upgrade. I assume that it has something to do with the SEARCH operator.

@kasakrisz Is this the same redundancy that you have been investigating?

Reply:
Yes, this behavior is introduced by https://issues.apache.org/jira/browse/CALCITE-4446